### PR TITLE
[WP#57970]Add warning when encryption for group folders are not enabled explicitly

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
This PR:
- Adds a warning in projectfolder setup UI when server-side encryption is enabled, but not specifically activated for group folders.
- Adds php unit test
- Adds vue ui test for it 

## Related Issue or Workpackage

- Fixes https://community.openproject.org/projects/nextcloud-integration/work_packages/57970/activity

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [x] Updated `CHANGELOG.md` file

### CODE REVIEW

File: src/components/AdminSettings.vue [hunk 1]
```diff
@@ -182,6 +182,7 @@
 				:title="t('integration_openproject', 'Project folders (recommended)')"
 				:is-setup-complete-without-project-folders="isSetupCompleteWithoutProjectFolders"
 				:is-there-error-after-project-folder-and-app-password-setup="isThereErrorAfterProjectFolderAndAppPasswordSetup"
+				:is-there-group-folders-encryption-warning="showEncryptionWarningForGroupFolders"
```
Comment (individual-it) [inline src/components/AdminSettings.vue:185]: any reason the names don't match each other?

File: src/components/AdminSettings.vue [hunk 2]
```diff
@@ -265,6 +266,12 @@
 						</p>
 						<p class="note-card--error-description" v-html="projectFolderSetUpErrorMessageDescription(state.project_folder_info.errorMessage)" /> <!-- eslint-disable-line vue/no-v-html -->
 					</NcNoteCard>
+					<NcNoteCard v-else-if="showEncryptionWarningForGroupFolders" class="note-card" type="warning">
+						<p class="note-card--title">
+							<b>{{ t('integration_openproject', 'Encryption for Group Folders are not enabled.') }}</b>
```
Comment (individual-it) [inline src/components/AdminSettings.vue:271]:
```suggestion
							<b>{{ t('integration_openproject', 'Encryption for the Group Folders App is not enabled.') }}</b>
```

File: src/components/AdminSettings.vue [hunk 3]
```diff
@@ -265,6 +266,12 @@
...
+						</p>
+						<p class="note-card--warning-description" v-html="getGroupFolderEncryptionWarningHint" /> <!-- eslint-disable-line vue/no-v-html -->
```
Comment (individual-it) [inline src/components/AdminSettings.vue:273]:
```suggestion
						<p class="note-card--warning-description" v-html="getGroupFoldersEncryptionWarningHint" /> <!-- eslint-disable-line vue/no-v-html -->
```

be consequent with singular/plural

File: src/components/AdminSettings.vue [hunk 4]
```diff
@@ -551,6 +558,11 @@ export default {
 			const hintTextForAdminAudit = t('integration_openproject', 'To activate audit logs for the OpenProject integration, please enable the {htmlLinkForAdminAudit} app and follow the configuration steps outlined in the {htmlLinkForDocumentaion}.', { htmlLinkForAdminAudit, htmlLinkForDocumentaion }, null, { escape: false, sanitize: false })
 			return dompurify.sanitize(hintTextForAdminAudit, { ADD_ATTR: ['target'] })
 		},
+		getGroupFolderEncryptionWarningHint() {
+			const linkText = t('integration_openproject', 'documentation')
+			const htmlLink = `<a class="link" href="https://www.openproject.org/docs/system-admin-guide/integrations/nextcloud/#files-are-not-encrypted-when-using-nextcloud-server-side-encryption" target="_blank" title="${linkText}">${linkText}</a>`
+			return t('integration_openproject', 'Server-side encryption is active, but encryption for group folders is not yet enabled. To ensure secure storage of files in project folders, please follow the configuration steps in the {htmlLink}.', { htmlLink }, null, { escape: false, sanitize: false })
```
Comment (individual-it) [inline src/components/AdminSettings.vue:564]: here `group folders` is written in lower case, please be consequent when referring to the Group Folders App

**Review by individual-it — CHANGES_REQUESTED** [review]

File: src/components/AdminSettings.vue (hunk 1 above)
Comment (SagarGi) [reply src/components/AdminSettings.vue:185]: Yes it can be changed according to the function name. `is-there-group-folders-encryption-warning`

**Review by individual-it — APPROVED** [review]

**Pull request merged**